Index: gst-plugins-base1.0/gst-libs/gst/audio/meson.build
===================================================================
--- gst-plugins-base1.0.orig/gst-libs/gst/audio/meson.build
+++ gst-plugins-base1.0/gst-libs/gst/audio/meson.build
@@ -153,7 +153,7 @@ gstaudio = library('gstaudio-@0@'.format
   soversion : soversion,
   darwin_versions : osxversion,
   install : true,
-  dependencies : gstaudio_deps,
+  dependencies : [ gstaudio_deps, base_platform_dep ],
 )
 pkg_name = 'gstreamer-audio-1.0'
 pkgconfig.generate(gstaudio,
Index: gst-plugins-base1.0/meson.build
===================================================================
--- gst-plugins-base1.0.orig/meson.build
+++ gst-plugins-base1.0/meson.build
@@ -399,6 +399,15 @@ else
   core_conf.set('DISABLE_ORC', 1)
 endif
 
+base_platform_dep = []
+# Check for libatomic for use of C11 atomics: some architectures need
+# to link explicitly to this library.
+# TODO: Once meson 1.6 is in use, this can be updated to
+# dependency('atomic')
+atomic_dep = cc.find_library('atomic', required: false)
+
+base_platform_dep += [atomic_dep]
+
 # Used to build SSE* things in audio-resampler
 sse_args = '-msse'
 sse2_args = '-msse2'
